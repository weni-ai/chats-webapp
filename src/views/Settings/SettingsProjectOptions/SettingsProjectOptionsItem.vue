<template>
  <section class="project-options__item">
    <UnnnicSwitch
      data-testid="settings-project-option-switch"
      :modelValue="modelValue"
      :textRight="name"
      @update:model-value="$emit('update:model-value', $event)"
    />
    <UnnnicToolTip
      v-if="tooltip"
      data-testid="settings-project-option-tooltip"
      enabled
      :text="tooltip"
      side="right"
      maxWidth="20rem"
    >
      <UnnnicIcon
        data-testid="settings-project-option-tooltip-icon"
        icon="info"
        scheme="neutral-soft"
        filled
        size="sm"
      />
    </UnnnicToolTip>
  </section>
</template>

<script>
export default {
  name: 'SettingsProjectOptionsItem',

  props: {
    modelValue: {
      type: Boolean,
      default: false,
    },
    name: {
      type: String,
      required: true,
    },
    tooltip: {
      type: String,
      default: '',
    },
  },

  emits: ['update:model-value'],
};
</script>

<style lang="scss" scoped>
.project-options__item {
  display: flex;
  gap: $unnnic-spacing-nano;
  align-items: center;

  .unnnic-tooltip {
    display: flex;
  }
}
</style>
