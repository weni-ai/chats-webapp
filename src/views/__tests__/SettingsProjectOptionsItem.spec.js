import { mount } from '@vue/test-utils';
import { describe, it, expect, beforeEach } from 'vitest';
import SettingsProjectOptionsItem from '@/views/Settings/SettingsProjectOptions/SettingsProjectOptionsItem.vue';

const createWrapper = (props) => {
  return mount(SettingsProjectOptionsItem, {
    props,
  });
};

describe('SettingsProjectOptionsItem.vue', () => {
  let wrapper;

  describe('Without tooltip', () => {
    beforeEach(() => {
      wrapper = createWrapper({ modelValue: true, name: 'Option Name' });
    });

    it('should render switch with correct props', () => {
      const switchComponent = wrapper.findComponent(
        '[data-testid=settings-project-option-switch]',
      );
      expect(switchComponent.exists()).toBe(true);
      expect(switchComponent.props('modelValue')).toBe(true);
      expect(switchComponent.props('textRight')).toBe('Option Name');
    });

    it('should not render tooltip when tooltip prop is not provided', () => {
      const tooltip = wrapper.findComponent(
        '[data-testid=settings-project-option-tooltip]',
      );
      expect(tooltip.exists()).toBe(false);
    });

    it('should emit update:model-value event when switch emits update:model-value', async () => {
      const switchComponent = wrapper.findComponent(
        '[data-testid=settings-project-option-switch]',
      );
      await switchComponent.vm.$emit('update:model-value', true);

      expect(wrapper.emitted()['update:model-value']).toBeTruthy();
      expect(wrapper.emitted()['update:model-value'][0]).toEqual([true]);
    });
  });

  describe('Without tooltip', () => {
    beforeEach(() => {
      wrapper = createWrapper({
        modelValue: true,
        name: 'Option Name',
        tooltip: 'This is a tooltip',
      });
    });

    it('should render tooltip when tooltip prop is provided', () => {
      const tooltip = wrapper.findComponent(
        '[data-testid=settings-project-option-tooltip]',
      );
      expect(tooltip.exists()).toBe(true);
      expect(tooltip.props('text')).toBe('This is a tooltip');
      expect(tooltip.props('side')).toBe('right');
    });

    it('should match the snapshot', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
