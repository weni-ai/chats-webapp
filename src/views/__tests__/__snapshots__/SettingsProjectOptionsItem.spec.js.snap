// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SettingsProjectOptionsItem.vue > Without tooltip > should match the snapshot 1`] = `
<section
  class="project-options__item"
  data-v-0e9bc71d=""
>
  <div
    class="unnnic-switch"
    data-testid="settings-project-option-switch"
    data-v-0b325129=""
    data-v-0e9bc71d=""
  >
    <!---->
    <svg
      class="unnnic-icon unnnic-icon__size--md unnnic--clickable unnnic-icon-scheme--brand-weni unnnic-switch__icon active"
      data-v-0b325129=""
      data-v-930380cc=""
      data-v-fbb3af00=""
      disabled="false"
      height="40"
      id="default"
      viewBox="0 0 80 40"
      width="80"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        class="primary"
        data-v-fbb3af00=""
        height="35.8"
        rx="17.9"
        width="71.6"
        x="4.20013"
        y="2.10046"
      />
      <circle
        cx="22.5"
        cy="20.0005"
        data-v-fbb3af00=""
        fill="white"
        id="default-circle"
        r="12.5"
      />
    </svg>
    <div
      class="unnnic-switch__label unnnic-switch__label__right unnnic-switch__label__medium"
      data-v-0b325129=""
    >
      Option Name
    </div>
  </div>
  <div
    class="unnnic-tooltip"
    data-testid="settings-project-option-tooltip"
    data-v-0e9bc71d=""
    data-v-5e81b292=""
  >
    
    <span
      class="material-symbols-rounded unnnic-icon-scheme--neutral-soft unnnic-icon-size--sm material-symbols-rounded--filled"
      data-testid="settings-project-option-tooltip-icon"
      data-v-0e9bc71d=""
      data-v-930380cc=""
    >
      info
    </span>
    
    <span
      class="unnnic-tooltip-label unnnic-tooltip-label-right"
      data-v-5e81b292=""
      style="max-width: 20rem; left: 8px; top: 0px;"
    >
      
      
      This is a tooltip 
      <br
        data-v-5e81b292=""
      />
      
      
      <!---->
    </span>
  </div>
</section>
`;
